### PR TITLE
Add GetChannelInformation endpoint

### PR DESCRIPTION
**Pull request by codedninja**
Added endpoint function for getting channel information, also cleared up some formatting and documentation.

### CODE REVIEW

**Comment by nicklaw5** [comment]
Thanks @codedninja. I'll take a look over the next day or so.

File: channels.go [hunk 1]
```diff
@@ -49,3 +49,43 @@ func (c *Client) SearchChannels(params *SearchChannelsParams) (*SearchChannelsRe
...
+	Channels []ChannelInformation `json:"data"`
+}
+
+// ChannelInformation ...
+type ChannelInformation struct {
+	ID         string `json:"broadcaster_id"`
+	Display    string `json:"broadcaster_name"`
+	Language   string `json:"broadcaster_language"`
+	CategoryID string `json:"game_id"`
+	Category   string `json:"game_name"`
```
Comment (nicklaw5) [inline channels.go:75]: Please adjust these field names as described below:

```suggestion
	BroadcasterID         string `json:"broadcaster_id"`
	BroadcasterName    string `json:"broadcaster_name"`
	BroadcasterLanguage   string `json:"broadcaster_language"`
	GameID string `json:"game_id"`
	GameName   string `json:"game_name"`
```

File: channels_test.go [hunk 2]
```diff
@@ -109,3 +109,89 @@ func TestSearchChannels(t *testing.T) {
...
+			&Options{ClientID: "my-client-id"},
+			"44445592",
+			`{"data":[{"broadcaster_id":"44445592","broadcaster_login":"pokimane","broadcaster_name":"pokimane","broadcaster_language":"en","game_id":"509658","game_name":"Just Chatting","title":"See you Wednesday 8am for Among Us ^_^"}]}`,
+			[]ChannelInformation{
+				{
+					ID:         "44445592",
+					Display:    "pokimane",
+					Language:   "en",
+					CategoryID: "509658",
+					Category:   "Just Chatting",
```
Comment (nicklaw5) [inline channels_test.go:134]: These will need to change also:

```suggestion
					BroadcasterID:         "44445592",
					BroadcasterName:    "pokimane",
					BroadcasterLanguage:   "en",
					GameID: "509658",
					GameName:   "Just Chatting",
```

File: channels_test.go [hunk 3]
```diff
@@ -109,3 +109,89 @@ func TestSearchChannels(t *testing.T) {
...
+		}
+
+		if resp.StatusCode != testCase.statusCode {
+			t.Errorf("expected status code to be \"%d\", got \"%d\"", testCase.statusCode, resp.StatusCode)
+		}
+
+		for i, channel := range resp.Data.Channels {
+			if channel.ID != testCase.parsed[i].ID {
+				t.Errorf("Expected struct field ID = %s, was %s", testCase.parsed[i].ID, channel.ID)
+			}
```
Comment (nicklaw5) [inline channels_test.go:174]:
```suggestion
			if channel.BroadcasterID != testCase.parsed[i].BroadcasterID {
				t.Errorf("Expected struct field BroadcasterID = %s, was %s", testCase.parsed[i].BroadcasterID, channel.BroadcasterID)
			}
```

File: channels_test.go [hunk 4]
```diff
@@ -109,3 +109,89 @@ func TestSearchChannels(t *testing.T) {
...
+
+			if channel.Display != testCase.parsed[i].Display {
+				t.Errorf("Expected struct field DisplayName = %s, was %s", testCase.parsed[i].Display, channel.Display)
+			}
```
Comment (nicklaw5) [inline channels_test.go:178]:
```suggestion
			if channel.BroadcasterName != testCase.parsed[i].BroadcasterName {
				t.Errorf("Expected struct field BroadcasterName = %s, was %s", testCase.parsed[i].BroadcasterName, channel.BroadcasterName)
			}
```

File: channels_test.go [hunk 5]
```diff
@@ -109,3 +109,89 @@ func TestSearchChannels(t *testing.T) {
...
+
+			if channel.Language != testCase.parsed[i].Language {
+				t.Errorf("Expected struct field Language = %s, was %s", testCase.parsed[i].Language, channel.Language)
+			}
```
Comment (nicklaw5) [inline channels_test.go:182]:
```suggestion
			if channel.BroadcasterLanguage != testCase.parsed[i].BroadcasterLanguage {
				t.Errorf("Expected struct field BroadcasterLanguage = %s, was %s", testCase.parsed[i].BroadcasterLanguage, channel.BroadcasterLanguage)
			}
```

File: channels_test.go [hunk 6]
```diff
@@ -109,3 +109,89 @@ func TestSearchChannels(t *testing.T) {
...
+
+			if channel.CategoryID != testCase.parsed[i].CategoryID {
+				t.Errorf("Expected struct field CategoryID = %s, was %s", testCase.parsed[i].CategoryID, channel.CategoryID)
+			}
```
Comment (nicklaw5) [inline channels_test.go:186]:
```suggestion
			if channel.GameID != testCase.parsed[i].GameID {
				t.Errorf("Expected struct field GameID = %s, was %s", testCase.parsed[i].GameID, channel.GameID)
			}
```

File: channels_test.go [hunk 7]
```diff
@@ -109,3 +109,89 @@ func TestSearchChannels(t *testing.T) {
...
+
+			if channel.Category != testCase.parsed[i].Category {
+				t.Errorf("Expected struct field Category = %s, was %s", testCase.parsed[i].Category, channel.Category)
+			}
```
Comment (nicklaw5) [inline channels_test.go:190]:
```suggestion
			if channel.GameName != testCase.parsed[i].GameName {
				t.Errorf("Expected struct field GameName = %s, was %s", testCase.parsed[i].GameName, channel.GameName)
			}
```

**Review by nicklaw5 — CHANGES_REQUESTED** [review]
@codedninja Looks good. Just a few minor field names I'd like changed. See below comments.

File: channels.go (hunk 1 above)
Comment (codedninja) [reply channels.go:75]: Reason why I changed it to `CategoryID` and `CategoryName` is due to them releasing the new EventSub API and it returning that same information as `Category`. Might be a sign in the near future they will change it to that on helix.

Comment (nicklaw5) [reply channels.go]: Right, thanks for explaining.However, until that change comes into effect I'd prefer we represent the API as it stands today.

Thanks for making those changes.

**Review by nicklaw5 — APPROVED** [review]
:+1:

**Pull request merged**